external/vcm/vcm/limit.py

```diff
@@ -0,0 +1,99 @@
+import xarray as xr
+from sklearn.base import BaseEstimator, TransformerMixin
+from typing import Optional, Sequence, Mapping
+
+
+class DatasetQuantileLimiter(BaseEstimator, TransformerMixin):
+    """Transformer to reduce the extremity of outliers of a dataset to quantile limits.
+    
+    Limiting is done on a variable by variable basis and along specified dimensions.
+        Limits are optionally computed on a configurable subset of the dataset to avoid
+        loading the entire dataset.
+    
+    Args:
+        alpha: two-tailed alpha for computing extrema quantiles, values beyond which
+            will be reduced to the quantile
+        limit_only: variables in the dataset to be limited; if not specified all
+            variables are limited
+
+    """
+
+    def __init__(
+        self, alpha: float, limit_only: Optional[Sequence[str]] = None,
```

Comment:
I suggest using two knobs for upper and lower bounds:

```suggestion
        self, lower_quantile_limit: float, upper_quantile_limit: float, limit_only: Optional[Sequence[str]] = None,
```

I could see wanting to use different quantiles for the upper and lower threshold. Also, its hard to tell from the docstring how `alpha` is turned into upper/lower bounds. Finally, the "alpha to upper/lower quantile" formula is in two places...the tests and this class, so increases the implicit coupling.